netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -155,6 +165,13 @@ public Runnable start(Listener transportListener) {
     if (NioSocketChannel.class.isAssignableFrom(channelType)) {
       b.option(SO_KEEPALIVE, true);
     }
+
+    /**
+     * Proxy connections needs to happen first, unencrypted, before any other operation.
+     */
+    if (proxyHandler != null) {
+      b.handler(proxyHandler);
```

Comment:
Ah, my bad! The least intrusive way I come up with was adding `ProtocolNegotiation#useProxy`. Let me know what you think.